src/app/menu/menu.component.ts

```diff
@@ -1,15 +1,28 @@
 import { Component, OnInit } from '@angular/core';
+import { Router } from '@angular/router';
+import { AuthService } from '../auth/services/auth.service';
 
 @Component({
   selector: 'app-menu',
   templateUrl: './menu.component.html',
   styleUrls: ['./menu.component.scss']
 })
 export class MenuComponent implements OnInit {
-
-  constructor() { }
+  pedidoAlmuerzo:Pedido[]=[];
+  constructor(private router: Router, private authService:AuthService) { }
 
   ngOnInit(): void {
   }
 
+  logout(){
+    this.authService.logout();
+    this.router.navigateByUrl('auth/login');
+  }
+
+  verpedidos(){
+    if(this.pedidoAlmuerzo.length > 0){
+      this.router.navigateByUrl('verpedidos');
```

Comment:
Deberías guardar el arreglo de platillos en el servicio, así para que el siguiente módulo lo ocupe.
En esta función haces:

this.menuService.guardarPlatillos(this.pedidoAlmuerzo);

// Servicio menú
pedido: Pedido[] = []; // Variable de clase.

// Función que solo guarda el pedido en el servicio.
guardarPlatillos(orden: Pedido[]) { 
  this.pedido = orden;
}

Con esto el módulo de Órdenes podrá acceder a los platillos seleccionados.